From c5ae8add47db5907883e2dcaf9e744f4d21b62a6 Mon Sep 17 00:00:00 2001
From: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Date: Tue, 22 Jun 2021 13:49:43 +0000
Subject: [PATCH 1615/1921] octeontx2-af: cn10k: config to identify ethertype
 location

Configure NIX_AF_RX_DEF_ET with the layer information, which will be
used to identify the ethertype location in L2 header.
The ethertype location or offset is used to overwrite the ethertype
field in L2 header in inlne IPsec flow.

Signed-off-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Change-Id: Ie409185d29f44047e9f25dd9b2934283a26cbdb2
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/55072
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index f8db5bb7cead..b6685c31e2d4 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -15521,7 +15521,9 @@ static struct npc_lt_def_cfg npc_lt_defaults = {
 	},
 	.rx_et = {
 		{
-			.lid = NPC_LID_LB,
+			.offset = -2,
+			.valid = 1,
+			.lid = NPC_LID_LC,
 			.ltype_match = NPC_LT_NA,
 			.ltype_mask = 0x0,
 		},
-- 
2.31.1

